### PR TITLE
Update metrics outputter to accept group as an optional column

### PR DESCRIPTION
**Pull request by yetamsft**
TODO: unit test the added scenario

### CODE REVIEW

File: assets/model_monitoring/components/src/model_monitor_metric_outputter/builder/metric_output_builder.py [hunk 1]
```diff
@@ -35,34 +35,31 @@ def _build(self, metrics: List[Row]) -> dict:
...
-                    group_names.append(node_id)
+            try:
+                metric_name = metric_dict[SIGNAL_METRICS_METRIC_NAME]
+
+                group_names = []
+                if SIGNAL_METRICS_GROUP in metric_dict:
+                    root_group = metric_dict[SIGNAL_METRICS_GROUP]
+                    group_names.append(root_group)
 
                     if metric_dict[SIGNAL_METRICS_GROUP_DIMENSION].lower() != AGGREGATE:
```
Comment (yetamsft) [inline assets/model_monitoring/components/src/model_monitor_metric_outputter/builder/metric_output_builder.py:47]: TODO: need to check whether GROUP_DIMENSION exists here. Currently, it throws KeyError, which is a bug.

Comment (chupadhy) [reply assets/model_monitoring/components/src/model_monitor_metric_outputter/builder/metric_output_builder.py]: Done

File: assets/model_monitoring/components/tests/unit/test_metric_output_builder.py [hunk 2]
```diff
@@ -77,6 +78,118 @@ def test_metrics_with_1_level_groups(self):
             },
         }
 
+    def test_empty_groups_and_group_dimensions_success(self):
```
Comment (chupadhy) [inline assets/model_monitoring/components/tests/unit/test_metric_output_builder.py:81]: This change will be merged as part of another PR
